lib/user-methods.js

```diff
@@ -92,32 +92,39 @@ function refreshAccessToken(user, localRealmPath, realmUrl) {
             // We do this because in React Native Remote Debugging
             // `Realm.clearTestState()` will have invalidated the user object
             let newUser = user.constructor.all[user.identity];
-            if (newUser) {
-                let session = newUser._sessionForOnDiskPath(localRealmPath);
-                if (session) {
-                    const errorHandler = session.config.error;
-                    if (response.status != 200) {
-                        let error = new AuthError(json);
-                        if (errorHandler) {
-                            errorHandler(session, error);
-                        } else {
-                            print_error('Unhandled session token refresh error', error);
-                        }
-                    } else if (session.state !== 'invalid') {
-                        parsedRealmUrl.set('pathname', json.access_token.token_data.path);
-                        session._refreshAccessToken(json.access_token.token, parsedRealmUrl.href);
-
-                        if (errorHandler && errorHandler._notifyOnAccessTokenRefreshed) {
-                            errorHandler(session, errorHandler._notifyOnAccessTokenRefreshed)
-                        }
-
-                        const tokenExpirationDate = new Date(json.access_token.token_data.expires * 1000);
-                        scheduleAccessTokenRefresh(newUser, localRealmPath, realmUrl, tokenExpirationDate);
-                    }
+            if (!newUser) {
+                return;
+            }
+            let session = newUser._sessionForOnDiskPath(localRealmPath);
+            if (!session) {
+                print_error(`Unhandled session token refresh error: could not look up session at path ${localRealmPath}`);
+            }
+
+            const errorHandler = session.config.error;
+            if (response.status != 200) {
+                let error = new AuthError(json);
+                if (errorHandler) {
+                    errorHandler(session, error);
                 } else {
-                    print_error(`Unhandled session token refresh error: could not look up session at path ${localRealmPath}`);
+                    print_error('Unhandled session token refresh error', error);
                 }
+                return;
+            }
+            if (session.state === 'invalid') {
+                return;
+            }
+
+            const tokenData = json.access_token.token_data;
+
+            parsedRealmUrl.set('pathname', tokenData.path);
+            session._refreshAccessToken(json.access_token.token, parsedRealmUrl.href, tokenData.sync_label);
```

Comment:
If I'm reading the diff correctly, the only relevant portion of this change is the addition of this argument here, with the remainder simply being improving the formatting?

Reply:
Yes. I started out thinking it needed significantly more changes, then it turned out that the data I needed was already there...